### ISSUE TITLE
doxygenclass option : display the list of inherited classe(s)

### ISSUE THREAD

**Issue by guyguy333** [michaeljones/breathe#220]
Hi,

Sometimes it's useful to display class inheritance, i.e. the list of inherited classe(s) (for explanations). 
However, the current version of breathe does not offer this feature.  Maybe this feature should be a doxygenclass option. 

What do you think ? It would be awesome to see this feature in a future release.

Regards

**Comment by michaeljones** [comment]
Hi,

Thanks for the suggestion. I would agree that it would make sense to provide that as an option. How would you go about presenting the information? Is there a format that you've seen in other tools that you like?

Cheers,
Michael

**Comment by guyguy333** [comment]
I think there are multiple options:

* Use doxygendiagram pending feature to display inheritance graph
* An indented item list with the name of each class including an internal link if the class is "defined" somewhere else.

Maybe it would be necessary to have an option to limit inheritance level. I mean, if A inherit of B which inherit of C maybe we only want to show that A inherit of B.

In parallel to this feature, sometimes it's useful to comment members of a class A inherited from the class B and not the class B. However, to document inherited members from class B, it's necessary to use class B syntax (B::member) which is not easily comprehensible if we don't specify that A inherit from B.
For example, it would be nice to have something like

.. doxygenfunction:: A::memberFromClassA
.. doxygenfunction:: A::memberInheritedFromClassB

Thank you,

Guillaume

**Comment by ela34** [comment]
In term of presentation, We should be happy with something as simple as the example below. 
Below the class description listing, we should have "inherits from" followed by a list (comma separated) of the base classes, Clicking one of the base class name should just jump to the class description.

#include “CoolFramework/TimerModule.h

class **awTimerModule**
A module to manage timers.
Inherits from **awBaseModule**

The same way around we could have in the base class, the list of class that inherits from it:

#include “CoolFramework/BaseModule.h

class **awBaseModule**
The base class for modules.
Subclassed by **awTimerModule**, **awToasterModule**

Do you think this makes sense and that it is feasible?

Thanks you.

-- Eric

**Comment by michaeljones** [comment]
Hi, I have just pushed a branch called `feat/inheritance` with the first version of support for this feature. Please take a look when you have time and report back any issues. Thanks for your patience and for the great suggestion!

**Comment by kratsg** [comment]
Hi @michaeljones what does the `feat/inheritance` do? I don't see a `:inherited-members:` option for the directive? I came here from #174 and may end up going by way of hand-crafting links to base classes for the user to click through to.

**Comment by michaeljones** [comment]
Hi,

`feat/inheritance` includes output of base & derived classes. The output isn't optional at this point but it doesn't take up too much space. The names are links to the classes themselves if they are in the docs somewhere.

Hope that helps!
Michael

**Comment by vitaut** [comment]
Closing this issue because Breathe renders the list of base classes now ([example](https://breathe.readthedocs.org/en/latest/class.html#qt-slots-example)):

![screenshot from 2016-03-30 11 17 55](https://cloud.githubusercontent.com/assets/576385/14152631/13e340c2-f669-11e5-81fe-1cf737d45f43.png)

**Comment by eric-wieser** [comment]
I've reimplemented this to use the builtin cpp domain mechanism in #295
